app/models/piece.rb

```diff
@@ -1,8 +1,121 @@
-class Piece < ApplicationRecord
-  belongs_to :user
-  belongs_to :game
-
-  def self.types
-    %w[King Queen Rook Bishop Knight Pawn]
-  end
-end
+
+class Piece < ApplicationRecord
+  belongs_to :user
+  belongs_to :game
+
+  def self.types
+    %w[King Queen Rook Bishop Knight Pawn]
+  end
+
+  def obstructed?(destination_x, destination_y)
+    raise 'Error: Invalid Input' if destination_y > 7 || destination_x > 7 || destination_y < 0 || destination_x < 0
+    destination_tests =
+      [
+        horizontal_obstruction_left?(destination_x, destination_y),
+        horizontal_obstruction_right?(destination_x, destination_y),
+        vertical_obstruction_up?(destination_x, destination_y),
+        vertical_obstruction_down?(destination_x, destination_y),
+        diagonal_obstruction_down_left?(destination_x, destination_y),
+        diagonal_obstruction_up_right?(destination_x, destination_y),
+        diagonal_obstruction_down_right?(destination_x, destination_y),
+        diagonal_obstruction_up_left?(destination_x, destination_y)
+      ]
+    raise 'Error: Invalid Input' if !destination_tests.include?(true) && !destination_tests.include?(false)

```

Comment:
@RaykPlatzek Here is something I thought about this morning that may work for us.  I wasn't at my computer so I wasn't able to run tests.

https://gist.github.com/roderickjoseph/3ef07506a3e63369a77b32824f5b7005